### PR TITLE
Add documentation for UI Test Automation Framework

### PR DESCRIPTION
**Pull request by OleksandrOrlov**
The PR adds the documentation for the new approach for the UI Automation Framework, that is already used for paritioner test modules like 'lvm', 'no_separate_home' etc.

The PR is in progress and added to collect comments, questions and ideas on the early stage.

To see the MD files with applied markdown please see them in my branch: https://github.com/OleksandrOrlov/os-autoinst-distri-opensuse/tree/46766_framework_documentation/ui-framework-documentation

### CODE REVIEW

File: ui-framework-documentation/layers.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+* [Validation Test Module](#validation-test-module)
+    * [Test Data usage in Validation Test Module](#test-data-usage-in-validation-test-module)
+    * [Access to os-autoinst testapi from Validation Test Module](#access-to-os-autoinst-testapi-from-validation-test-module)
+    * [Access to other framework layers from Validation Test Module](#access-to-other-framework-layers-from-validation-test-module)
+
+    
+## Test Module
+
+Test Module is a layer containing test case steps that need to be
+executed on the system under test.
```
Comment (jknphy) [inline ui-framework-documentation/layers.md]: (SUT) and use it in the rest of the document.

File: ui-framework-documentation/layers.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+But in case if need to verify some negative cases the method cannot be
+used as is. For instance, to verify that prompt appears when blank
+password is entered. For such kind of cases the access to the
+[Pages](#page) through [Controller](#controller) is added.  
+Then [Test Module](#test-module) code may look like:
+
+```perl
+sub run {
+    my $partitioner = $testapi::distri->get_partitioner();
```
Comment (jknphy) [inline ui-framework-documentation/layers.md]: perhaps to put all together we might want to name controlers? for example get_partitioner_controller? Page concept seems more clear to me than Controller in the examples. make sense?

File: ui-framework-documentation/layers.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+Test Module is a layer containing test case steps that need to be
+executed on the system under test.
+
+### Test Data usage in Test Module
+
+All the data for the test should be provided on this level. Do not
+provide any test data in [Controller](#controller) or [Page](#page) 
+layers.
+
```
Comment (jknphy) [inline ui-framework-documentation/layers.md]: some example of data?

File: ui-framework-documentation/layers.md (hunk 2 above)
Comment (OleksandrOrlov) [reply ui-framework-documentation/layers.md]: Yes, makes sense.  Also, for the particular case, it probably should even be something like 'edit_suggested_partitioning_controller', as we also have Expert Partitioner.

Naming part is still under consideration. For example, I'm not sure what is better name for controller layer - "Controller" or, for example "Steps".

File: ui-framework-documentation/layers.md (hunk 1 above)
Comment (OleksandrOrlov) [reply ui-framework-documentation/layers.md]: Thank you. I also considered this abbreviation. As it well-known for everyone it makes sense to shorten the occurrences. Will change it.

File: ui-framework-documentation/layers.md (hunk 3 above)
Comment (OleksandrOrlov) [reply ui-framework-documentation/layers.md]: Ok. Will add an example.

File: ui-framework-documentation/layers.md (hunk 1 above)
Comment (rwx788) [reply ui-framework-documentation/layers.md]: Usually you should introduce it first and then use.

Comment (OleksandrOrlov) [reply ui-framework-documentation/layers.md]: Did not update it yet, but will do this soon.

**Comment by jknphy** [comment]
In general I like how is explained, I only found some difficulty to connect everything, like for someone not so used to this pattern (I will include myself partially), to see all the calls, like selenium does for the typical login/home page. It doesn't need to be existing code, but I ami imaging something like a sequence of simple code: (1) factory pattern to get the distri (2) couple of method calling testapi function to type in a simple page (invented) for the Page (3) Controller methods (4) test data and calls to the page methods. Basically and **all-together at the end of the document**, so however want to try will know that creating this 4 files just can do something and then the test will go more in depth with this doc for specific things that are recommended or not.
I think for someone to embrace this good idea it could be needed some basic example. WDYT? Another option would be not included in the doc and do some sample code and push it to the repo that it would help as a template for anyone.

**Comment by OleksandrOrlov** [comment]
Thanks for the feedback. This it what I'm looking for. 

Agree. It will add the examples.

**Pull request merged**